parser/parser.go

```diff
@@ -379,133 +388,88 @@ func (p *parser) evaluateOperator(value1, value2 Expression, operator rune,
 	}, nil
 }
 
-func (p *parser) addMaps(map1, map2 []*Property, pos scanner.Position) ([]*Property, error) {
-	ret := make([]*Property, 0, len(map1))
-
-	inMap1 := make(map[string]*Property)
-	inMap2 := make(map[string]*Property)
-	inBoth := make(map[string]*Property)
-
-	for _, prop1 := range map1 {
-		inMap1[prop1.Name] = prop1
-	}
-
-	for _, prop2 := range map2 {
-		inMap2[prop2.Name] = prop2
-		if _, ok := inMap1[prop2.Name]; ok {
-			inBoth[prop2.Name] = prop2
-		}
-	}
-
-	for _, prop1 := range map1 {
-		if prop2, ok := inBoth[prop1.Name]; ok {
-			var err error
-			newProp := *prop1
-			newProp.Value, err = p.evaluateOperator(prop1.Value, prop2.Value, '+', pos)
-			if err != nil {
-				return nil, err
-			}
-			ret = append(ret, &newProp)
-		} else {
-			ret = append(ret, prop1)
-		}
-	}
-
-	for _, prop2 := range map2 {
-		if _, ok := inBoth[prop2.Name]; !ok {
-			ret = append(ret, prop2)
-		}
-	}
-
-	return ret, nil
-}
-
-func (p *parser) parseOperator(value1 Expression) *Operator {
-	operator := p.tok
-	pos := p.scanner.Position
-	p.accept(operator)
-
-	value2 := p.parseExpression()
-
-	value, err := p.evaluateOperator(value1, value2, operator, pos)
-	if err != nil {
-		p.error(err)
-		return nil
-	}
-
-	return value
-
-}
-
 func (p *parser) parseValue() (value Expression) {
 	switch p.tok {
 	case scanner.Ident:
-		return p.parseVariable()
+		switch text := p.scanner.TokenText(); text {
```

Comment:
My thoughts about whether to inline some functions since I know that a discussion about this is happening soon:

Either approach here seems reasonable to me. Since they're still clearly delineated by the case statements and likely to only be used once anyway I think it's ok to inline them, but as long as the function names are clear (which I think they are) I think also think it's easy to read them as separate functions.